Shared/Samples/Control annotation sublayer visibility/ControlAnnotationSublayerVisibilityView.Model.swift

```diff
@@ -0,0 +1,126 @@
+            annotationLayer.subLayerContents[0] as! AnnotationSublayer
+        }
+        
+        /// The open annotation sublayer.
+        private var openSublayer: AnnotationSublayer {
+            annotationLayer.subLayerContents[1] as! AnnotationSublayer
+        }
+        
+        /// A Boolean value indicating whether to show the closed annotation sublayer.
+        @Published var showsClosedSublayer = true
+        
+        /// A Boolean value indicating whether to show the open annotation sublayer.
+        @Published var showsOpenSublayer = true
+        
+        /// A Boolean value indicating whether the open annotation sublayer is visible at the current extent.
+        @Published var visibleAtCurrentExtent = false
+        
+        /// The min scale of the open annotation sublayer.
+        private var minScale: Double?
+        
+        /// The max scale of the open annotation sublayer.
+        private var maxScale: Double?
+        
+        /// The current scale of the map.
+        var currentScale: Double = .zero {
+            didSet {
+                formatCurrentScaleText()
+                guard minScale != nil && maxScale != nil else { return }
+                visibleAtCurrentExtent = openSublayer.isVisible(atScale: currentScale)
+            }
+        }
+        
+        /// The map's current scale value in text.
+        @Published var currentScaleText = ""
+        
+        /// The min and max scale of the open annotation layer in text.
+        @Published var minMaxScaleText = ""
+        
+        /// Loads a local mobile map package.
+        func loadMobileMapPackage() async throws {
+            mobileMapPackage = MobileMapPackage(fileURL: .gasDevicePackage)
+            try await mobileMapPackage.load()
+            guard let map = mobileMapPackage.maps.first else { return }
+            self.map = map
+        }
+        
+        /// Sets the visibility of the closed annotation sublayer.
+        /// - Parameter visibility: The visibility of the sublayer.
+        func setClosedSublayerVisibility(_ visibility: Bool) {
+            closedSublayer.isVisible = visibility
+        }
+        
+        /// Sets the visibility of the open annotation sublayer.
+        /// - Parameter visibility: The visibility of the sublayer.
+        func setOpenSublayerVisibility(_ visibility: Bool) {
+            openSublayer.isVisible = visibility
+        }
+        
+        /// Sets the min and max scale and formats the min-max scale text.
+        func setScaleText() {
+            minScale = openSublayer.minScale
+            maxScale = openSublayer.maxScale
+            
+            formatMinMaxScaleText()
+        }
+        
+        /// Formats the current scale text.
+        private func formatCurrentScaleText() {
+            currentScaleText = "1:\(currentScale.formatted(.decimal))"
+        }
+        
+        /// Formats the min-max scale text.
+        private func formatMinMaxScaleText() {
```

Comment:
I'm not sure this function carries it's own weight as it's only ever called once that I can see.